zarr/src/codecs/bytes_to_bytes.ml

```diff
@@ -13,10 +14,13 @@ module GzipCodec = struct
       Error (Printf.sprintf "Invalid Gzip level %d" i)
 
   let encode l x =
-    Ezgzip.compress ~level:(to_int l) x
+    Bytes.Reader.to_string @@
+    Bytesrw_zlib.Gzip.compress_reads ~level:(to_int l) () @@
+    Bytes.Reader.of_string x
 
   let decode x =
-    Result.get_ok @@ Ezgzip.decompress x
+    Bytes.Reader.to_string @@
+    Bytesrw_zlib.Gzip.decompress_reads () @@ Bytes.Reader.of_string x
```

Comment:
@dbuenzli Am I using `bytesrw_zlib` and `bytesrw_zstd` correctly here? I noticed a slight slowdown in speed of the testsuite. Maybe there is something I can do to maximize performance of (de)compression give that im using a string (that'l not be used again) as input?

Reply:
Are you sure this is the culprit ? Looks ok to me.  

Normally `Bytes.Reader.of_string` [does not perform](https://github.com/dbuenzli/bytesrw/blob/517d1685b031f38b354fb348e42b15b4251a6a58/src/bytesrw.ml#L468-L470) a copy. Now [`Bytes.Reader.to_string`](https://github.com/dbuenzli/bytesrw/blob/517d1685b031f38b354fb348e42b15b4251a6a58/src/bytesrw.ml#L511-L513) allocates a `Buffer`, so this may entail a bit of overhead. If you are repeatedly doing so in a tight loop you may want to allocate a single one and use `Bytes.Reader.add_to_buffer` instead.

Reply:
Okay that makes sense. Thanks!

Reply:
It turns out the slowdown of the testsuite was simply due to the addition of Zstd unit tests and nothing else. `Ezgzip.decompress` and `Bytesrw_zlib` have no significant performance differences.

Reply:
Ah thanks :–)